packages/presentation/lib/main_bloc.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:bloc/bloc.dart';
+import 'package:domain/usecase/game_state.dart';
+import 'package:domain/usecase/generate_number_usecase.dart';
+import 'package:domain/usecase/check_state_usecase.dart';
+import 'package:flutter/material.dart';
+import 'main_state.dart';
+import 'main_event.dart';
+
+class MainBloc extends Bloc<MainEvent, MainState> {
+  final GenerateNumberUseCase generateNumberUseCase;
+  final CheckStateUseCase checkStateUseCase;
+  final myTextController = TextEditingController();
+  bool? isGuessed;
+  int number = 1;
+  int randomNumber = 1;
+  int counter = 0;
+
+  MainBloc(
+      {required this.generateNumberUseCase, required this.checkStateUseCase})
+      : super(MainInitial()) {
+    on<CheckNumberEvent>((event, emit) {
+      emit(_checkNumber());
+    });
+    on<NewGameEvent>((event, emit) {
+      emit(_newGame());
+    });
+  }
+
+  void generateRandomNumber() {
+    randomNumber = generateNumberUseCase();
+  }
+
+  MainState _checkNumber() {
+    number = int.tryParse(myTextController.text)!;
+    final params = ComparedNumbers(
+      guessNumber: number,
+      randomNumber: randomNumber,
+      counter: counter++,
+    );
+    if (checkStateUseCase(params) is OutOfAttempts) {
+      return OutOfAttemptsState();
+    }
+    if (checkStateUseCase(params) is YouWon) {
+      return YouWonState();
+    }
```

Comment:
extract checkStateUseCase to variable and then work with a variable in conditions. and change flow from 
if ...
if ...
return

to
if
else if
else (or return)